`PYTHON.json` object file annotations for 3.11 aren't aware of frozen module changes

Python 3.11 introduced a multi-phase build approach where some additional .pyc are frozen into the binary. The build system changes appear to have invalidated some of our object file annotation assumptions. This is causing linker errors with the 3.11 distributions on PyOxidizer. e.g.

```
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(pylifecycle.o): in function `finalize_interp_clear':
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           pylifecycle.c:(.text+0x1cd9): undefined reference to `_Py_Deepfreeze_Fini'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(pylifecycle.o): in function `pycore_interp_init':
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           pylifecycle.c:(.text+0x21c6): undefined reference to `_Py_Deepfreeze_Init'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0x18): undefined reference to `_Py_get_importlib__bootstrap_toplevel'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0x38): undefined reference to `_Py_get_importlib__bootstrap_external_toplevel'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0x58): undefined reference to `_Py_get_zipimport_toplevel'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0x98): undefined reference to `_Py_get_abc_toplevel'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0xb8): undefined reference to `_Py_get_codecs_toplevel'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0xd8): undefined reference to `_Py_get_io_toplevel'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0xf8): undefined reference to `_Py_get__collections_abc_toplevel'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0x118): undefined reference to `_Py_get__sitebuiltins_toplevel'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0x138): undefined reference to `_Py_get_genericpath_toplevel'
[2023-01-16T17:07:14Z WARN  pyoxidizer::project_building]           /usr/bin/ld: /home/runner/work/PyOxidizer/PyOxidizer/target/debug/tempdir/pyoxidizerYN643l/build/target/x86_64-unknown-linux-gnu/debug/deps/libpyo3_ffi-a5bfc332461a2de5.rlib(frozen.o):(.data.rel.ro+0x158): undefined reference to `_Py_get_ntpath_toplevel'
```

I want to say this regression is only limited to the PYTHON.json build metadata on the 3.11 distributions. So I'm not going to make it a release blocker. But it does prevent the adoption of 3.11 distributions in PyOxidizer until we have a fix.

--- a/cpython-unix/build-cpython.sh
+++ b/cpython-unix/build-cpython.sh
@@ -830,7 +830,7 @@ ${BUILD_PYTHON} ${ROOT}/fix_shebangs.py ${ROOT}/out/python/install
 
 # Also copy object files so they can be linked in a custom manner by
 # downstream consumers.
-OBJECT_DIRS="Objects Parser Parser/pegen Programs Python"
+OBJECT_DIRS="Objects Parser Parser/pegen Programs Python Python/deepfreeze"
 OBJECT_DIRS="${OBJECT_DIRS} Modules"
 for ext in _blake2 cjkcodecs _ctypes _ctypes/darwin _decimal _expat _hacl _io _multiprocessing _sha3 _sqlite _sre _xxtestfuzz ; do
     OBJECT_DIRS="${OBJECT_DIRS} Modules/${ext}"

